test/hotspot/jtreg/vmTestbase/nsk/share/jdi/JDIBase.java

```diff
@@ -150,6 +152,36 @@ protected final void getEventSet() throws JDITestRuntimeException {
         }
     }
 
+    // Special version of getEventSet for ThreadStartEvent/ThreadDeathEvent.
+    // When ThreadStartRequest and/or ThreadDeathRequest are enabled, we can get the events from system threads
+    // unexpected for tests.
+    // The method skips ThreadStartEvent/ThreadDeathEvent events for all threads except the expected one.
+    protected void getEventSetForThreadStartDeath(String threadName) throws JDITestRuntimeException {
+        boolean gotDesiredEvent = false;
+        while (!gotDesiredEvent) {
+            getEventSet();
+            Event event = eventIterator.nextEvent();
+            if (event instanceof ThreadStartEvent evt) {
+                if (evt.thread().name().equals(threadName)) {
+                    gotDesiredEvent = true;
+                } else {
+                    log2("Got ThreadStartEvent for wrong thread: " + event);
+                }
+            } else if (event instanceof ThreadDeathEvent evt) {
+                if (evt.thread().name().equals(threadName)) {
+                    gotDesiredEvent = true;
+                } else {
+                    log2("Got ThreadDeathEvent for wrong thread: " + event);
+                }
+            } else {
+                // not ThreadStartEvent nor ThreadDeathEvent
+                gotDesiredEvent = true;
```

Comment:
Do you think it would be better if you only set `gotDesiredEvent = true` if the event came in on the named thread? If the event came in on another thread, it seems it should be ignored. If it came in on the named thread, then it is probably an error since the test is expecting ThreadStart or ThreadDeath, so in that case return and let the caller complain.

Reply:
Not sure I got the idea.
gotDesiredEvent is a flag to exit the cycle, so if the event is not ThreadStart/Death event, we set the flag, leave the cycle and return the event to the caller. In case of request001 test it's an error and the test will fail, but this is desired behaviour

Reply:
The purpose is to ignore all events on other threads (not just ThreadStart and ThreadDeath) rather than let result in the test failing. However, I can see now that determining the thread of the event is not easy (lots of instanceOf checks of the event type and casts to the proper type), so probably this is not worth doing unless it proves necessary at some point.